pandas/core/arrays/base.py

```diff
@@ -1433,7 +1435,14 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
                 f"'{type(self).__name__}' with dtype {self.dtype} "
                 f"does not support reduction '{name}'"
             )
-        return meth(skipna=skipna, **kwargs)
+        result = meth(skipna=skipna, **kwargs)
+
+        if keepdims:
+            # if subclasses want to avoid wrapping in np.array, do:
+            # super()._reduce(..., keepdims=False) and wrap that.
+            return np.array([[result]])
```

Comment:
should we warn/deprecate here?

Reply:
This reminds me that using `*args` & `**kwargs` in the method signature makes changing APIs difficult, in addition to them making typing more difficult. I really think they should be avoided whenever possible.

Is there a good/clean way to add a deprecation? I assume if any user has been using a `keepdims` parameter in their code, it's likely been a bool, so using type checking for deprecation inside `_reduce` would fail in those cases. We could check `kwargs` in `blk_func` in `frame.py` for key `"keepdims"`. That depends on us not calling `_reduce(keepdims)` in any other location (we probably won't) or remembering to add deprecation texts there also. Any better idea?